include/wil/wistd_functional.h

```diff
@@ -297,28 +297,34 @@ namespace wistd     // ("Windows Implementation" std)
       template <class _Fp>
       using _EnableIfCallable = typename enable_if<__callable<_Fp>::value>::type;
     public:
-        typedef _Rp result_type;
+        using result_type = _Rp;
 
         // construct/copy/destroy:
         __WI_LIBCPP_INLINE_VISIBILITY __WI_LIBCPP_SUPPRESS_NONINIT_ANALYSIS
-        function() WI_NOEXCEPT : __f_(0) {}
+        function() WI_NOEXCEPT : __buf_(), __f_(0)
+        {
+        }
+
         __WI_LIBCPP_INLINE_VISIBILITY
-        function(nullptr_t) WI_NOEXCEPT : __f_(0) {}
+        function(nullptr_t) WI_NOEXCEPT : __buf_(), __f_(0)
+        {
+        }
+
         function(const function&);
-        function(function&&);
+        function(function&&) WI_NOEXCEPT;

```

Comment:
There's no guarantee that the underlying function type is `noexcept` movable (same thing with `swap`)

Reply:
No. Move constructors can throw